src/gluonts/mx/distribution/transformed_distribution.py

```diff
@@ -51,6 +51,10 @@ def __init__(
         self._event_shape: Optional[Tuple] = None
         self._batch_shape: Optional[Tuple] = None
 
+    @property
+    def support(self) -> Tuple[Tensor, Tensor]:
+        return self.base_distribution.support
```

Comment:
I think this should be something like:

```python
lb, ub = self.base_distribution.support
for t in self.transforms:
  lb = t.f(lb)
  ub = t.f(ub)
return lb, ub
```

Reply:
This was my original implementation but there was a minor issue with that so I reverted it to the version I have now. 

The issue is that the support in most of the distributions is defined in the "base" distribution part and in theory the affine transformation should be applied to it as well to scale it appropriately. However, in the binned distribution there is effectively no base distribution and the support is defined on the already scaled version. This causes a discrepancy when we have for example a mixture since the final support of distributions that in theory should have the same support does not match since in one case the affine transformation is applied and in the other case not. 

Also, since binned uses this support range for the scaled version, it raises the question if we should actually scale the support or not. For example for a gaussian distribution the support is the real line regardless of the scale, so here I basically make the assumption that the support range that such distributions have should not be scaled since it resembles a "fake infinity". 

For all the above I decided to just propagate the support as is to the transformed distribution, making implicitly the assumption that the limits of the support (e.g. `1e16`) apply to both base and scaled distributions since they are effectively equivalent to the real line. If I was using for example `np.fifo(np.float64).max` as limit, scaling would be effectively useless. I made this assumption but for lower limits.

Note that this assumption is problematic for other cases like uniform (this is the only example I could think of) where the support should be actually scaled since it does not extend to "fake infinity". However, I could not think of a clean solution for everything so I would appreciate feedback here. Should we actually set it to infinity and cut it to a proper value when need to use it?

Reply:
I'm not sure I understand the issue here, but I think this might be related to https://github.com/awslabs/gluon-ts/issues/1050 (which I think would simplify a few things).

> This causes a discrepancy when we have for example a mixture since the final support of distributions that in theory should have the same support does not match since in one case the affine transformation is applied and in the other case not.

Why should they have the same support?

Reply:
@vafl I think it's hard to get the support for transformed variables under general transformations: even if one is interested in the inf and sup of the support, you probably need to restrict yourself to monotonic transformations. Or am I missing something? This is why I believe that focusing on affine transformations might be better

Reply:
on the other hand, bijections are fine (as long as they are continuous)